js_modules/dagit/packages/core/src/partitions/AssetJobPartitionsView.tsx

```diff
@@ -0,0 +1,139 @@
+        ]),
+      ),
+    [assetHealth, partitionNames],
+  );
+
+  const [pageSize, setPageSize] = React.useState(60);
+  const [offset, setOffset] = React.useState<number>(0);
+  const [showSteps, setShowSteps] = React.useState(false);
+
+  React.useEffect(() => {
+    if (viewport.width && !showSteps) {
+      // magical numbers to approximate the size of the window, which is calculated in the step
+      // status component.  This approximation is to make sure that the window does not jump as
+      // the pageSize gets recalculated
+      const _approximatePageSize = Math.ceil((viewport.width - 330) / 32) - 3;
+      setPageSize(_approximatePageSize);
+    }
+  }, [viewport.width, showSteps, setPageSize]);
+
+  const selectedPartitions = showSteps
+    ? partitionNames.slice(
+        Math.max(0, partitionNames.length - 1 - offset - pageSize),
+        partitionNames.length - offset,
+      )
+    : partitionNames;
+
+  return (
+    <div>
+      <Box
+        flex={{justifyContent: 'space-between', direction: 'row', alignItems: 'center'}}
+        border={{width: 1, side: 'bottom', color: Colors.KeylineGray}}
+        padding={{vertical: 16, horizontal: 24}}
+      >
+        <Subheading>Status</Subheading>
+        <Box flex={{gap: 8}}>
+          <Button onClick={() => setShowSteps(!showSteps)}>
+            {showSteps ? 'Hide per-asset status' : 'Show per-asset status'}
+          </Button>
+          <LaunchAssetExecutionButton
+            context="all"
+            assetKeys={assetGraph.graphAssetKeys}
+            preferredJobName={pipelineName}
+          />
+        </Box>
+      </Box>
+      <Box
+        flex={{direction: 'row', alignItems: 'center'}}
+        border={{width: 1, side: 'bottom', color: Colors.KeylineGray}}
+        padding={{left: 8}}
+      >
+        <CountBox count={partitionNames.length} label="Total partitions" />
+        <CountBox
+          count={partitionNames.filter((x) => jobHealth[x] === PartitionState.MISSING).length}
+          label="Missing partitions"
+        />
+      </Box>
+      <Box padding={{vertical: 16, horizontal: 24}}>
+        <div {...containerProps}>
+          <PartitionStatus
+            partitionNames={partitionNames}
+            partitionData={jobHealth}
+            selected={showSteps ? selectedPartitions : undefined}
+            selectionWindowSize={pageSize}
+            onClick={(partitionName) => {
+              const maxIdx = partitionNames.length - 1;
+              const selectedIdx = partitionNames.indexOf(partitionName);
+              const nextOffset = Math.min(
+                maxIdx,
+                Math.max(0, maxIdx - selectedIdx - 0.5 * pageSize),
+              );
+              setOffset(nextOffset);
+              if (!showSteps) {
+                setShowSteps(true);
+              }
+            }}
+            tooltipMessage="Click to view per-step status"
+          />
+        </div>
+        {showSteps && <Box margin={{top: 16}} />}
```

Comment:
is the per-step status missing here?

How are we populating the per-asset state? I assume it's fetching the full asset status (materialization counts by partition) rather than filtering by last run?

Reply:
Hey hey! Ahh sorry I should have clarified, I split the per-asset status out into the second PR that is stacked on this one (https://github.com/dagster-io/dagster/pull/10385) because it required a lot of work.

I think the `showSteps` should be renamed to showAssets though.

We're populating the per-asset state using materializationCountByPartition + the batch loader I added in https://github.com/dagster-io/dagster/pull/10382. 

We're not filtering by last run but I think that's ok? The goal of this work is to remove run / op-centric partition state so that this page shows asset partition status that matches what you see when you view individual asset pages or go to launch asset backfills. (I think?)